src/stores/widgets/CatalogWidgetStore.ts

```diff
@@ -64,22 +81,432 @@ export class CatalogWidgetStore {
-        this.catalogColor = DEFAULTS.catalogColor;
-        this.catalogSize = DEFAULTS.catalogSize;
-        this.catalogShape = DEFAULTS.catalogShape;
-        this.xAxis = DEFAULTS.xAxis;
-        this.yAxis = DEFAULTS.yAxis;
+        this.headerTableColumnWidts = [150, 75, 65, 100, 230];
+        this.showSelectedData = false;
+        this.catalogTableAutoScroll = false;
+        this.catalogPlotType = CatalogPlotType.ImageOverlay;
+        this.catalogColor = Colors.TURQUOISE3;
+        this.catalogSize = 10;
+        this.catalogShape = CatalogOverlayShape.CircleLined;
+        this.xAxis = CatalogOverlay.NONE;
+        this.yAxis = CatalogOverlay.NONE;
         this.tableSeparatorPosition = PreferenceStore.Instance.catalogTableSeparatorPosition;
-        this.highlightColor = DEFAULTS.highlightColor;
-        this.settingsTabId = DEFAULTS.settingsTabId;
+        this.highlightColor = Colors.RED2;
+        this.settingsTabId = CatalogSettingsTabs.SIZE;
+        this.thickness = 2.0;
+        this.sizeMapColumn = CatalogOverlay.NONE;
+        this.sizeArea = false;
+        this.sizeScalingType = FrameScaling.LINEAR;
+        this.sizeMin = {area: 100, diameter: 5};
+        this.sizeMax = {area: 200, diameter: 20};
+        this.sizeColumnMin = {default: undefined, clipd: undefined};
+        this.sizeColumnMax = {default: undefined, clipd: undefined};
+        this.sizeAxisTabId = CatalogSettingsTabs.SIZE_MAJOR;
+        this.sizeMinorMapColumn = CatalogOverlay.NONE;
+        this.sizeMinorArea = false;
+        this.sizeMinorScalingType = FrameScaling.LINEAR;
+        this.sizeMinorColumnMin = {default: undefined, clipd: undefined};
+        this.sizeMinorColumnMax = {default: undefined, clipd: undefined};
+        this.colorMapColumn = CatalogOverlay.NONE;
+        this.colorColumnMax = {default: undefined, clipd: undefined};
+        this.colorColumnMin = {default: undefined, clipd: undefined};
+        this.colorMap = "viridis";
+        this.colorScalingType = FrameScaling.LINEAR;
+        this.invertedColorMap = false;
+        this.orientationMapColumn = CatalogOverlay.NONE;
+        this.orientationMax = {default: undefined, clipd: undefined};
+        this.orientationMin = {default: undefined, clipd: undefined};
+        this.orientationScalingType = FrameScaling.LINEAR;
+        this.angleMax = 360;
+        this.angleMin = CatalogWidgetStore.MinAngle;
+        this.sizeColumnMinLocked = false;
+        this.sizeColumnMaxLocked = false;
+        this.sizeMinorColumnMinLocked = false;
+        this.sizeMinorColumnMaxLocked = false;
+
+        reaction(() => this.sizeMapData, (column) => {
+            const result = minMaxArray(column);
+            this.setSizeColumnMin(isFinite(result.minVal)? result.minVal : 0, "default");
+            this.setSizeColumnMax(isFinite(result.maxVal)? result.maxVal : 0, "default");
+        });
+
+        reaction(() => this.sizeArray(), (size) => {
```

Comment:
I'm confused. How you can you react to a simple function that returns a `Float32Array`. Should `sizeArray` be a `@computed` function?

Reply:
I think that because we have a `@computed` function inside `sizeArray`. with `@computed sizeArray`, we get a render bug when user pre set size map.